pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/executor/RoundRobinScheduler.java

```diff
@@ -79,128 +69,145 @@ public RoundRobinScheduler(long releaseTimeout) {
   public RoundRobinScheduler(long releaseTimeoutMs, Supplier<Long> ticker) {
     _releaseTimeout = releaseTimeoutMs;
     _ticker = ticker;
+    _availableOpChainReleaseService = Executors.newSingleThreadScheduledExecutor(r -> {
+      Thread t = new Thread(r);
+      t.setName(AVAILABLE_RELEASE_THREAD_NAME);
+      t.setDaemon(true);
+      return t;
+    });
+    if (releaseTimeoutMs > 0) {
+      _availableOpChainReleaseService.scheduleAtFixedRate(() -> {
+        for (Map.Entry<OpChainId, Long> entry : _available.entrySet()) {
```

Comment:
note: Iterating over a map and deleting is error-prone even with concurrent hash-map. I'll split this into different loops.